### PR TITLE
WIP - optimize go2make in module mode

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

go2make currently tries to use package traversal expressions that cross modules and use vendored paths. it also makes use of `go/build` to load package info, which is multiple orders of magnitude slower when building in module mode.

This PR is hoisted out of https://github.com/kubernetes/kubernetes/pull/99226 to solve the following issues in go2make:
* adjust package expressions to point to valid go packages
* optimize package loading time, even when run in gomodule mode

xref https://github.com/kubernetes/kubernetes/issues/82531

```release-note
NONE
```

### CODE REVIEW

File: build/root/Makefile.generated_files [hunk 1]
```diff
@@ -68,11 +68,11 @@ $(META_DIR)/$(GO_PKGDEPS_FILE): FORCE
...
 	fi
 	hack/run-in-gopath.sh go install ./hack/make-rules/helpers/go2make
-	hack/run-in-gopath.sh go2make                       \
-	    k8s.io/kubernetes/...                           \
-	    --prune  k8s.io/kubernetes/staging              \
-	    --prune  k8s.io/kubernetes/vendor               \
-	    k8s.io/kubernetes/vendor/k8s.io/...             \
+	hack/run-in-gopath.sh go2make                                                                  \
+	    k8s.io/kubernetes/{build,cmd,pkg,plugin,test}/...                                          \
+	    $(shell ls staging/src/k8s.io | xargs printf -- "k8s.io/kubernetes/vendor/k8s.io/%s/... ") \
```
Comment (dims) [inline build/root/Makefile.generated_files:73]: whoa. hadn't seen that before. good one :) `xargs printf` with `%s`

**Comment by thockin** [comment]
Modules mode is/was VERY BROKEN when trying to evaluate multiple packages.  I'll take a look at this, but the last I saw it, we  were waiting on several deep bugs in go tooling to be resolved

**Comment by liggitt** [comment]
yeah, hold... sorry, was still stacking PRs to make sure this works in both modes

**Pull request closed without merging**